src/ripple/protocol/impl/STLedgerEntry.cpp

```diff
@@ -76,15 +76,19 @@ void STLedgerEntry::setSLEType ()
         Throw<std::runtime_error> ("invalid ledger entry type");
 
     type_ = mFormat->getType ();
-    if (!setType (mFormat->elements))
+    try
     {
-        if (auto j = debugJournal().warn())
+        setType(mFormat->elements);
+    }
+    catch(std::exception&)
```

Comment:
Hmm we should not be using exceptions to indicate errors arising from user inputs. To be clear, these are examples of allowed uses of exceptions:
* A required file is unavailable
* An internal logic error is detected
* The configuration file has a syntax error

These are examples where throwing an exception should be avoided:
* A network message is invalid
* A client submits an object that cannot be deserialized

What the disallowed cases have in common, is the principle that untrusted inputs should not be allowed to make the program generate exceptions at-will.